Update dependencies in package-lock.json

pfefferle opened the pull request:
Upgraded several dependencies including @tanstack/react-router, baseline-browser-mapping, body-parser, cookie, cookie-signature, finalhandler, hashery, http-errors, ipaddr.js, node-forge, preact, qs, raw-body, react-day-picker, react-remove-scroll, send, and others to their latest versions. Also removed or added 'peer' fields in some packages and updated sub-dependencies to address security, compatibility, and feature improvements.

<!--- Provide a general summary of your changes in the Title above -->

Fixes #

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
*

### Other information:

- [ ] Have you written new tests for your changes, if applicable?

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Go to '..'
*

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger a GitHub workflow that will create and push the entry into the branch. -->

<!-- Due to org permissions, the job may fail for PRs crated from a fork under GitHub organizations. -->
<!-- In this case, you can create entry manually with `composer changelog:add` and push it into the branch. -->

<!-- If no changelog entry is required for this PR, please add the "Skip Changelog" label. -->

-   [ ] Automatically create a changelog entry from the details below.

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Added - for new features
-   [ ] Changed - for changes in existing functionality
-   [ ] Deprecated - for soon-to-be removed features
-   [ ] Removed - for now removed features
-   [ ] Fixed - for any bug fixes
-   [ ] Security - in case of vulnerabilities

#### Message

<!-- Add a changelog message here -->

</details>

pfefferle committed:
Update dependencies in package-lock.json

Upgraded several dependencies including @tanstack/react-router, baseline-browser-mapping, body-parser, cookie, cookie-signature, finalhandler, hashery, http-errors, ipaddr.js, node-forge, preact, qs, raw-body, react-day-picker, react-remove-scroll, send, and others to their latest versions. Also removed or added 'peer' fields in some packages and updated sub-dependencies to address security, compatibility, and feature improvements.

obenland commented:
Does running `npm install` locally remove all these peer things for you? It adds it for me. I keep wondering why we get different results for that.

Which version of npm are you using? I'm at 11.6.2

pfefferle commented:
I ran an update...

pfefferle committed:
Merge branch 'trunk' into update/npm

obenland commented:
yeah same here but it restores the `peer: true` annotations for me.

What's your local npm version `npm -v`?

pfefferle commented:
10.8.3

pfefferle commented:
I meant I ran `npm update` not `npm install`!

obenland commented:
> 10.8.3

That might explain it. Is there anything keeping you on 10 or could you update to latest? `npm install -g npm@latest`

pfefferle committed:
Update browserslist and related dependencies

Upgraded browserslist to 4.28.1 and update-browserslist-db to 1.2.0 in package-lock.json, along with related dependency versions for improved compatibility and security.

pfefferle commented:
Same with `11.6.4`

pfefferle commented:
It seem to have nothing to do with the version.
